nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/controller/ControllerFacade.java

```diff
@@ -564,6 +577,62 @@ private boolean implementsServiceType(final Class serviceType, final Class type)
     }
 
+
+    /**
+     * Gets the RuntimeManifest for this overall NiFi instance.
+     *
+     * @return the runtime manifest
+     */
+    public RuntimeManifest getRuntimeManifest() {
+        final ExtensionManager extensionManager = getExtensionManager();
+        final Set<Bundle> allBundles = extensionManager.getAllBundles();
+
+        final Bundle frameworkBundle = NarClassLoadersHolder.getInstance().getFrameworkBundle();
+        final BundleDetails frameworkDetails = frameworkBundle.getBundleDetails();
+        final Date frameworkBuildDate = frameworkDetails.getBuildTimestampDate();
+
+        final BuildInfo buildInfo = new BuildInfo();
+        buildInfo.setVersion(frameworkDetails.getCoordinate().getVersion());
+        buildInfo.setRevision(frameworkDetails.getBuildRevision());
+        buildInfo.setCompiler(frameworkDetails.getBuildJdk());
+        buildInfo.setTimestamp(frameworkBuildDate == null ? null : frameworkBuildDate.getTime());
+
+        final RuntimeManifestBuilder manifestBuilder = new StandardRuntimeManifestBuilder()
+                .identifier("nifi")
+                .runtimeType("nifi")
+                .version(buildInfo.getVersion())
+                .schedulingDefaults(SchedulingDefaultsFactory.getNifiSchedulingDefaults())
+                .buildInfo(buildInfo);
+
+        for (final Bundle bundle : allBundles) {
+            getExtensionManifest(bundle).ifPresent(em -> manifestBuilder.addBundle(em));
+        }
+
+        return manifestBuilder.build();
+    }
+
+    private Optional<ExtensionManifest> getExtensionManifest(final Bundle bundle) {
+        final BundleDetails bundleDetails = bundle.getBundleDetails();
+        final File manifestFile = new File(bundleDetails.getWorkingDirectory(), "META-INF/docs/extension-manifest.xml");
+        if (!manifestFile.exists()) {
+            logger.warn("Unable to find extension manifest for [{}] at [{}]...", bundleDetails.getCoordinate(), manifestFile.getAbsolutePath());
+            return Optional.empty();
+        }
+
+        try (final InputStream inputStream = new FileInputStream(manifestFile)) {
+            final ExtensionManifest extensionManifest = extensionManifestParser.parse(inputStream);
+            // Newer NARs will have these fields populated in extension-manifest.xml, but older NARs will not, so we can
+            // set the values from the BundleCoordinate which already has the group, artifact id, and version
+            extensionManifest.setGroupId(bundleDetails.getCoordinate().getGroup());
+            extensionManifest.setArtifactId(bundleDetails.getCoordinate().getId());
+            extensionManifest.setVersion(bundleDetails.getCoordinate().getVersion());
+            return Optional.of(extensionManifest);
+        } catch (final IOException e) {
+            logger.error("Unable to load extension manifest for: " + bundleDetails.getCoordinate(), e);
```

Comment:
This should be changed to use placeholder variables:
```suggestion
            logger.error("Unable to load extension manifest for bundle {}", bundleDetails.getCoordinate(), e);
```